src/components/landing/pages/register/register.tsx

```diff
@@ -26,22 +26,15 @@ export const Register: React.FC = () => {
   const [error, setError] = useState(RegisterError.NONE)
   const [ready, setReady] = useState(false)
 
-  const doAsyncRegister = async (): Promise<void> => {
-    await doInternalRegister(username, password)
-    await getAndSetUser()
-  }
-
-  const doRegisterSubmit = (event: FormEvent): void => {
-    doAsyncRegister().catch((err: Error) => {
-      if (err.message === RegisterError[RegisterError.USERNAME_EXISTING]) {
-        setError(RegisterError.USERNAME_EXISTING)
-      } else {
-        setError(RegisterError.OTHER)
-      }
+  const doRegisterSubmit = useCallback((event: FormEvent) => {
+    doInternalRegister(username, password).then(async () => {
+      await getAndSetUser()
```

Comment:
You don't need to define a new async function. `getAndSetUser` already returns a promise.

Reply:
You're right.

Reply:
Done